onnxruntime/core/framework/graph_partitioner.h

```diff
@@ -51,7 +56,8 @@
                    const logging::Logger& logger,
                    Mode mode = Mode::kNormal,
                    const EpContextModelGenerationOptions& ep_context_gen_options = {},
-                   const layout_transformation::DebugGraphFn& debug_graph_fn = {}) const;
+                   const layout_transformation::DebugGraphFn& debug_graph_fn = {},
+                   const OnPartitionAssignmentFunction& on_partition_assign_fn = {}) const;
```

Comment:
>const OnPartitionAssignmentFunction& on_partition_assign_fn =  [](http://example.com/codeflow?start=19&length=62)

nit: it seems like the function signature is growing. Would you consider making it a member of the class?
Functions can be be null or non-null.